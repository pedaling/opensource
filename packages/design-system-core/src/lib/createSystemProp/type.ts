import type { Theme } from '../types';

export type InterpolationFn = (props: Record<string, any>) => Record<string, any>;

export type SystemPropConfig = {
  property: string;
  styleProperty?: string;
  scale?: keyof Pick<Theme, 'colors' | 'space' | 'opacity'>;
  disabled?: boolean;
  transform?: (value: any) => Record<string, any> | undefined;
};

export type SystemProp = {
  (props: Record<string, any>): Record<string, any>;
  propName: string;
  disabled: boolean;
};
