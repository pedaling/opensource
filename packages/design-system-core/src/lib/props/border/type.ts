import type { ResponsiveValue } from '../../types';

export type BorderProps = {
  borderWidth?: ResponsiveValue<number>;
  borderStyle?: ResponsiveValue<'solid' | 'none'>;
  borderColor?: ResponsiveValue<string>;
  borderTopWidth?: ResponsiveValue<number>;
  borderTopStyle?: ResponsiveValue<'solid' | 'none'>;
  borderTopColor?: ResponsiveValue<string>;
  borderRightWidth?: ResponsiveValue<number>;
  borderRightStyle?: ResponsiveValue<'solid' | 'none'>;
  borderRightColor?: ResponsiveValue<string>;
  borderBottomWidth?: ResponsiveValue<number>;
  borderBottomStyle?: ResponsiveValue<'solid' | 'none'>;
  borderBottomColor?: ResponsiveValue<string>;
  borderLeftWidth?: ResponsiveValue<number>;
  borderLeftStyle?: ResponsiveValue<'solid' | 'none'>;
  borderLeftColor?: ResponsiveValue<string>;
  borderRadius?: ResponsiveValue<number | string>;
  borderTopLeftRadius?: ResponsiveValue<number | string>;
  borderTopRightRadius?: ResponsiveValue<number | string>;
  borderBottomLeftRadius?: ResponsiveValue<number | string>;
  borderBottomRightRadius?: ResponsiveValue<number | string>;
  outlineWidth?: ResponsiveValue<number>;
  outlineStyle?: ResponsiveValue<'solid' | 'none'>;
  outlineColor?: ResponsiveValue<string>;
  outlineOffset?: ResponsiveValue<number>;
};
