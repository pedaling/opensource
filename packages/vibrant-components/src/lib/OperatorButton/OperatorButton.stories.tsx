import type { ComponentMeta, ComponentStory } from '@storybook/react';
import { Icon } from '@vibrant-ui/icons';
import { OperatorButton } from './OperatorButton';

export default {
  title: 'OperatorButton',
  component: OperatorButton,
  args: {
    IconComponent: Icon.Add.Regular,
  },
} as ComponentMeta<typeof OperatorButton>;

export const Basic: ComponentStory<typeof OperatorButton> = props => <OperatorButton {...props} />;
