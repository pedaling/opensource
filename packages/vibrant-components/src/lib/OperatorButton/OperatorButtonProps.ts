import type { ComponentType } from 'react';
import { propVariant, withVariation } from '@vibrant-ui/core';
import type { IconProps } from '@vibrant-ui/icons';

export type OperatorButtonProps = {
  disabled?: boolean;
  IconComponent: ComponentType<IconProps>;
  onClick?: () => void;
};

export const withOperationButtonVariation = withVariation<OperatorButtonProps>()(
  propVariant({
    props: [
      {
        name: 'disabled',
        default: false,
        keep: true,
      },
    ],
    variants: {
      true: {
        backgroundColor: 'disable',
        pressable: false,
        iconFill: 'onView3',
      },
      false: {
        backgroundColor: 'surface1',
        pressable: { overlayColor: 'onView1', interactions: ['hover', 'focus', 'active'] },
        iconFill: 'onView1',
      },
    } as const,
  })
);
