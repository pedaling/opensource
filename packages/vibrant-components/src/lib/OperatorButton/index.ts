export { OperatorButton } from './OperatorButton';
export type { OperatorButtonProps } from './OperatorButtonProps';
