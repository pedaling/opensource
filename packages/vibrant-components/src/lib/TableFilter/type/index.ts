export type StringFilterOperator = 'contains' | 'empty' | 'equals' | 'notContains' | 'notEmpty' | 'notEquals';

export type StringFilter = {
  value: string;
  operator: StringFilterOperator[];
};

export type Filter = StringFilter & { dataKey: string };
