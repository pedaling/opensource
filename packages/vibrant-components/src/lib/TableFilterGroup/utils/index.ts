import type { DateFilterOperator, MultiSelectFilterOperator, Option, StringFilterOperator } from '../types';

export const isValueRequiredOperator = (op: string) => op === 'empty' || op === 'notEmpty';

export const isStringFilterValid = (filter: { value: string; operator: StringFilterOperator }) =>
  Boolean(filter.value) || isValueRequiredOperator(filter.operator);

export const isDateFilterValid = (filter: { value: Date[]; operator: DateFilterOperator }) => {
  if (isValueRequiredOperator(filter.operator)) return true;

  if (filter.operator === 'between') return filter.value.length >= 2;

  return filter.value.length >= 1;
};

export const isMultiSelectFilterValid = (filter: { value: Option['value'][]; operator: MultiSelectFilterOperator }) =>
  Boolean(filter.value.length > 0) || isValueRequiredOperator(filter.operator);
