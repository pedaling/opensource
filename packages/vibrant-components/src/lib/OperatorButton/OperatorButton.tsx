import { Box } from '@vibrant-ui/core';
import { Pressable } from '../Pressable';
import { withOperationButtonVariation } from './OperatorButtonProps';

export const OperatorButton = withOperationButtonVariation(
  ({ IconComponent, backgroundColor, iconFill, ...restProps }) => (
    <Box
      base={Pressable}
      p={7}
      borderWidth={0}
      borderRadius={2}
      backgroundColor={backgroundColor}
      overlayColor="onView1"
      interactions={['hover', 'focus', 'active']}
      {...restProps}
    >
      <IconComponent size={16} fill={iconFill} />
    </Box>
  )
);
