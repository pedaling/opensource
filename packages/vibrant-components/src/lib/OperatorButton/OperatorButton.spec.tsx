import { waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { Icon } from '@vibrant-ui/icons';
import type { ReactRenderer } from '@vibrant-ui/utils/testing';
import { createReactRenderer } from '@vibrant-ui/utils/testing';
import { OperatorButton } from './OperatorButton';

describe('<OperatorButton />', () => {
  const { render } = createReactRenderer();

  let renderer: ReactRenderer;
  let element: HTMLElement;
  let mockOnClick: jest.Mock;

  beforeEach(() => {
    mockOnClick = jest.fn();
  });

  describe('when OperatorButton rendered', () => {
    beforeEach(() => {
      renderer = render(
        <OperatorButton data-testid="operatorButton" IconComponent={Icon.Add.Regular} onClick={mockOnClick} />
      );

      element = renderer.getByTestId('operatorButton');
    });

    describe('when button clicked', () => {
      beforeEach(async () => {
        await waitFor(() => userEvent.click(element));
      });

      it('onClick called', () => {
        expect(mockOnClick).toBeCalled();
      });
    });

    it('match snapshot', () => {
      expect(renderer.container).toMatchSnapshot();
    });
  });
});
