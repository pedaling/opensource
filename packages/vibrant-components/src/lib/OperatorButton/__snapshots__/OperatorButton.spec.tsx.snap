// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OperatorButton /> when OperatorButton rendered match snapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  padding: 7px;
  border-width: 0;
  border-radius: 2px;
  background-color: rgba(0, 0, 0, 0.05);
  cursor: pointer;
  position: relative;
  overflow-x: hidden;
  overflow-y: hidden;
  z-index: 0;
}

.emotion-0::before {
  content: "";
  position: absolute;
  z-index: -1;
  left: 0;
  right: 0;
  top: 0;
  bottom: 0;
  opacity: 0;
  background-color: #101010;
}

@media (hover: hover) {
  .emotion-0:hover::before {
    content: "";
    opacity: 0.08;
  }
}

.emotion-0:active::before {
  content: "";
  opacity: 0.16;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 16px;
  height: 16px;
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-self: flex-start;
  -ms-flex-item-align: flex-start;
  align-self: flex-start;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
  width: 100%;
  height: 100%;
  fill: #101010;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-content: stretch;
  -ms-flex-line-pack: stretch;
  align-content: stretch;
  -webkit-flex-shrink: 1;
  -ms-flex-negative: 1;
  flex-shrink: 1;
}

<div>
  <button
    class="emotion-0"
    data-testid="operatorButton"
  >
    <span
      class="emotion-1"
    >
      <svg
        class="emotion-2"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          class="emotion-3"
          d="M21.25 10.75h-8v-8c0-.15-.1-.25-.25-.25h-2c-.15 0-.25.1-.25.25v8h-8c-.15 0-.25.1-.25.25v2c0 .15.1.25.25.25h8v8c0 .15.1.25.25.25h2c.15 0 .25-.1.25-.25v-8h8c.15 0 .25-.1.25-.25v-2c0-.15-.1-.25-.25-.25Z"
        />
      </svg>
    </span>
  </button>
</div>
`;
